### ISSUE TITLE
Invisible Collisions On Child Entities

### ISSUE THREAD

**Issue by Tourist50** [tsunamayo/Starship-EVO#2133]
### Short Description of the issue:

This seems to be a similar issue as something reported about two weeks ago; child entities seem to be clipping with bricks outside of the physical hitbox of the brick.

Original post about child entities colliding with bricks in strange ways: 
[https://github.com/tsunamayo/Starship-EVO/issues/1960](url)

### Steps to reproduce the issue:

Load in my blueprints, then hit the switch behind the hinges to toggle the cockpit open and closed. When it closes, it should get stuck on the wedges in front where it looks like nothing is colliding. It's a little tempermental, so you may have to open and close it a few times until it doesn't close properly.

### Environment
- Starship EVO version: 20w21c
- Graphics Card: Nvidia GTX 1070

### Additional information:

The first time I encountered this collision:
![image](https://user-images.githubusercontent.com/33591296/82710400-c8e52a80-9c50-11ea-816f-55be0a97e8c3.png)

Showing that it's these wedges at the front by removing them and having it work correctly:
![image](https://user-images.githubusercontent.com/33591296/82710453-e87c5300-9c50-11ea-885e-24117057b0a9.png)

Collision points on:
![image](https://user-images.githubusercontent.com/33591296/82710483-fb8f2300-9c50-11ea-8c5a-c6281218ed7c.png)

Recreating without cockpit blocks:
![image](https://user-images.githubusercontent.com/33591296/82710509-0a75d580-9c51-11ea-96f2-e8f6024c69b1.png)

Recreating on a different testbed with slightly different slopes:
![image](https://user-images.githubusercontent.com/33591296/82710544-18c3f180-9c51-11ea-9787-d070c3463028.png)

The two blueprints I've recreated this on:

[Cockpit Clipping Blueprints.zip](https://github.com/tsunamayo/Starship-EVO/files/4670460/Cockpit.Clipping.Blueprints.zip)

Please also include the `output_log.txt` file. You can find your log here: `%UserProfile%\AppData\LocalLow\Moonfire Entertainment\Starship EVO\`

[output_log.txt](https://github.com/tsunamayo/Starship-EVO/files/4670451/output_log.txt)

**Comment by tsunamayo** [comment]
Just to confirm the issue:
This is what you expect:
![image](https://user-images.githubusercontent.com/6604318/82716765-266e8b00-9cd4-11ea-980f-74d9c963281a.png)
this is what you get sometime:
![image](https://user-images.githubusercontent.com/6604318/82716794-469e4a00-9cd4-11ea-8308-9a6661297884.png)
I just want to confirm that the first picture is the correct behavior for you, because this cockpit mechanism is never gonna retract more than that.
Thanks

**Comment by Tourist50** [comment]
Initially I expected the cockpit to close all the way as sometimes it would after freshly loading the craft or rebuilding the hinge system. However, after a few open and close attempts it would eventually get stuck on what seemed like thin air.

**Comment by ItsCrusade** [comment]
It looks like the bottom lip of the canopy is colliding with the bricks it's trying to move past - this makes sense, as when the canopy is moving - the bottom lip is lower than the hinges rotation point, so to complete the rotation past the bricks, it (the lip of the canopy) has to move slightly forward - colliding with the bricks instead.

Yes it will work sometimes, but this is due to the collision check not catching that very small collision on the frames it's calculating.

**Comment by tsunamayo** [comment]
Yes @ItsCrusade is correct, there is small threshold to not get stuck too easily, but this mechanism could not close in real life. 
It is still stopping too soon though, so there is an issue still. I keep you posted.

**Comment by Majikmonster** [comment]
I have a similar issue to this with my own canopy. Sometimes, when I close it, it doesn't close all the way. Trying to reset it results in the same stuck behavior for maybe a dozen, 2 dozen tries before it closes all the way again. However, if it closes all the way the first time, then it doesn't have the problem for a similar number of tries.
![20200530223315_1](https://user-images.githubusercontent.com/51390907/83343288-41d62900-a2c6-11ea-83d7-9627e273e8e3.jpg)
A successful closing:
![20200530223302_1](https://user-images.githubusercontent.com/51390907/83343292-4bf82780-a2c6-11ea-9bcd-720c6f439a77.jpg)
An unsuccessful closing:
![20200530223455_1](https://user-images.githubusercontent.com/51390907/83343297-54e8f900-a2c6-11ea-8223-4d231b73c646.jpg)
The ship blueprint:
[Majikmonster_ship_police_patrol_hot fuzz.zip](https://github.com/tsunamayo/Starship-EVO/files/4707019/Majikmonster_ship_police_patrol_hot.fuzz.zip)
game version: 20w22d
[output_log.txt](https://github.com/tsunamayo/Starship-EVO/files/4707014/output_log.txt)
